### ISSUE TITLE
Refactor pages for Login.gov SSO flow

### ISSUE THREAD

**Issue by thekaveman** [cal-itp/benefits#363]
Date: March 2022
Deciders: @angela-tran @machikoyasuda @vykster @thekaveman
**Priority**: [No. 3 per the Roadmap](https://docs.google.com/document/d/1IFoa8Ye0IXwGXXwxFjMrm1s3617Dbv6l-E-aCB0kgnA/edit#)

## Background

The current identity solution does not support a number of key requirements for this application, including full translation into at least English and Spanish.

This epic is centered around implementing an alternate flow utilizing the Login.gov Single Sign-On (SSO) system.

### Feature flags

Unlike the work surrounding Courtesy Cards, this effort is focused on the existing DMV / senior eligibility flow. Yet much of the work of #317 has been completed and shipped to `dev`, so this builds on top of it nonetheless.

New features introduced as part of #317 are mostly transparent to the existing flow if left unconfigured. We will approach this set of features the same way.

## Definition of Done

The Benefits app is live, with the DMV login system having been replaced by the Login.gov login system.

## User Stories

* [x] #354
* [x] #355
* [x] #356
* [x] #357
* [x] #358
* [x] #359

## Tasks

* [ ] https://github.com/cal-itp/benefits/issues/621
* [ ] #615
* [x] #636
---
* [x] #423 
* [x] #372
* [x] #374
* [x] #370
* [x] #369
* [x] #315
* [x] #380
* [x] #433
* [x] #367
* [x] #365
* [x] #435 
* [x] #432  
* [x] #376
* [x] #456
* [x] #465
* [x] #466
* [x] #428  Status: @angela-tran PR open #516
* [x] Finalize all copy that needs to be done before launch. Get it into a place where nondevs can see it
* [x] #533 
* [x] #540 
---
* [x] #375
* [x] #440
* [x] #442 
* [x] #368
* [x] #371
* [x] #476
* [x] #446
* [x] #377 @angela-tran (PR #452)
* [x] #702
* [ ] **Security / vulnerability testing** #521 #672 
* [x] **Design QA** Testing full functionality with emphasis on design
* [x] **Copy QA** presentation layer testing can be done at this point: Testing copy in English and Spanish only.
* [x] **Product QA** Testing full functionality

### Log in flow

*What happens if the user is already logged in on Login.gov in the same browser session?*

* Does not matter, they still have to login again for this app

*What happens if the user clicks Back after going through the Login.gov flow?*

* Need to learn more about how the "login success" message is passed back to our app from Login.gov

### Log out flow

* Clicking log out button in the middle of the flow puts user back on ~`core:agency_index`~ `eligibility:start`
* Log out watcher (for browser close) should watch all pages after log in, not just `enrollment:success`
* The [`unload` event should not be used](https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event#usage_notes)
  * Use [`Document.visibilitychange`](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event), or 
  * [`Window.pagehide`](https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event) instead

### Help page

* Combine Payment Options and Help pages
* Currently, the Help page uses `viewmodel` and `page.html`, to be more flexible we need to change this to have its own template (payment options already takes this approach)
* Simplest option is the one where we just add sections/paragraphs to the Help page (no table of contents nav, no modals)

### Eligibility Start page

*How to handle 3 items?*

* Add new attribute(s) on verifier model (hard-coded count of items vs. supporting N items with a `ManyToMany`)
* Template always renders the bank card item, no need to model on verifier
* Whether verifier has 2 or 3 items total, always render using new dashed lines approach

*How to render dashed horizontal lines for 3 items?*

* Login.gov design system has something called a “process list”: https://design.login.gov/components/process-list/ 
* CA Web Template has “progress blocks”: https://template.webstandards.ca.gov/sample/progress-blocks.html

**Comment by afeld** [comment]
[Writeup about the architecture history, decisions, and open questions](https://docs.google.com/document/d/13EWBeCQtVUIYMnUOKiIsPPN-P83YHbDKjSEJxq292Hc/edit#)

**Comment by vykster** [comment]
Link to privacy policy will need to be updated. @thekaveman for process would you prefer I add as a task above and create an issue?

**Comment by thekaveman** [comment]
@vykster 👍 #636

**Comment by vykster** [comment]
@thekaveman would you be ok with me renaming this to better align the roadmap considering we are splitting out older adults with and without a SSN?

**Comment by afeld** [comment]
This has a lot of overlap with https://github.com/cal-itp/benefits/issues/363; is there a benefit to keeping both open, or can we close one?

**Comment by thekaveman** [comment]
@afeld I think you meant #855 ☝️?

This is #363.

**Comment by afeld** [comment]
D'oh, sorry, I meant https://github.com/cal-itp/benefits/issues/635.

**Comment by afeld** [comment]
With blessing from @vykster in https://github.com/cal-itp/benefits/issues/855#issuecomment-1211015108, closing this in favor of https://github.com/cal-itp/benefits/issues/635.
